### ISSUE TITLE
Console font bug, cursor is off

### ISSUE THREAD

**Issue by clankill3r** [benfry/processing4#19]
I reported a similar issue a while ago for the text editor but I noticed it happens in the console too.

![2019-11-06 09 57 22](https://user-images.githubusercontent.com/738650/68283583-15ce2e80-007c-11ea-8037-21cba2759ab8.gif)

I'm using the latest version of Samuel's website (which is from some time back, so it might been have fixed).

**Comment by sampottinger** [comment]
Hey there! The text editing area was resolved in https://github.com/sampottinger/processing/issues/103 (Processing's custom text area UI was not compatible with Java 11 possibly b/c JEP 263) and should be active in your build. I wasn't able to reproduce but I'll keep digging. Which font and at what size?

**Comment by sampottinger** [comment]
Hey there! Sorry... I was able to reproduce the console issue with Source Code Pro 14px. This is a bit funky because  I thought Processing simply used `DefaultStyledDocument` default drawing in `EditorConsole` unlike `JEditTextArea` in which processing uses custom drawing behavior. I'll need to do some digging... That said, the text edit area should be correct.

**Comment by clankill3r** [comment]
I had Source Code Pro 16px on OSX.

Op wo 6 nov. 2019 om 23:48 schreef A Samuel Pottinger <
notifications@github.com>:

**Comment by sampottinger** [comment]
Hey there! So... we aren't doing custom rendering for that component... It's just a standard JTextPane (see EditorConsole). I'm slightly worried this might be https://bugs.openjdk.java.net/browse/JDK-8158370 (see also https://bugs.openjdk.java.net/browse/JDK-8156217). :-/ The resolution would be wait for Java to fix upstream or override drawing behavior. I'll be back with more soon.

**Comment by sampottinger** [comment]
Well... I'm not quite sure how this bug is manifesting but I traced it down to https://github.com/processing/processing4/blob/master/app/src/processing/app/ui/EditorConsole.java#L173. It appears that there's a rounding error in the text size calculation (inside Swing) but it only happens for certain gutter sizes. Using the default gutter size or switching Editor.LEFT_GUTTER to basically anything other than 44px seems to fix it. ☹️ So that's not great. I tried a bunch of different font / font sizes and this seems to resolve. I don't think I want to take over rendering behavior so I'll put this up to a PR. As with sampottinger/processing#103 this doesn't seem to happen with most fonts but it does happen with Source Code Pro. 😢
